containers/src/Kokkos_DualView.hpp

```diff
@@ -788,11 +787,16 @@ class DualView : public ViewTraits<DataType, Properties...> {
   /// data as modified.
   template <class Device>
   void modify() {
+    if (modified_flags.data() == nullptr) return;
+
     if constexpr (impl_dualview_is_single_device) {
+      // We can't distinguish so mark both as modified
+      ++modified_flags(0);
+      impl_report_host_modification();
+      ++modified_flags(1);
+      impl_report_device_modification();
```

Comment:
With this change, doesn't `need_sync<[Host|Device]>()` always returns true while both `need_sync_host()` and `need_sync_device()` return false?

Reply:
Good point!